src/components/apps/listing/Listing.js

```diff
@@ -109,28 +97,61 @@ function Listing({
         isFetching: appInCategoryStatus,
         error: appsInCategoryError,
     } = useQuery({
-        queryKey: appsInCategoryKey,
+        queryKey: [
+            APPS_IN_CATEGORY_QUERY_KEY,
+            {
+                systemId: category?.system_id,
+                rowsPerPage,
+                orderBy,
+                order,
+                appTypeFilter: filter?.name,
+                page,
+                categoryId: category?.id,
+                userId: userProfile?.id,
+            },
+        ],
         queryFn: getAppsInCategory,
         config: {
-            enabled: appsInCategoryQueryEnabled,
+            enabled:
+                // Disable the query if the category ID is fake and the user is
+                // logged in.  The Navigation component should update the ID to
+                // the real ID.
+                category?.system_id &&
+                category?.id &&
+                (!userProfile?.id ||
+                    ![constants.APPS_UNDER_DEV, constants.FAV_APPS].includes(
+                        category?.id
+                    )),
```

Comment:
Adding the check if the user is logged in and also checking the category ID is the only functional change to whether this query is enabled.
This is a bit hacky, but I'm disabling this query when the user is logged in and has one of these "fake" category UUIDs, carried over from the anonymous user, in order for the `Navigation` component to have an opportunity to update the currently selected category with the real category UUID for that user.